GitCommands/Git/ExecutableExtensions.cs

```diff
@@ -200,9 +199,12 @@ string ComposeOutput()
 
                 // Invoke batch progress callback
                 action?.Invoke(new BatchProgressEventArgs(item.BatchItemsCount, result));
-            }
 
-            return result;
+                if (!result)
+                {
+                    throw new ExternalOperationException(arguments: item.Argument);
```

Comment:
One can specify an error message by providing an inner exception:
`throw new ExternalOperationException(arguments: item.Argument, innerException: new Exception("Git command failed."));`

Open items:
1. There are ongoing discussions regarding localization (in non-UI modules).
2. Errors from git commands should be reported centrally #6527.

Reply:
Good idea. With an inner exception with the output of stderr is definitely better.

Reply:
![image](https://user-images.githubusercontent.com/9248729/110252285-f2bfd180-7f84-11eb-994f-0f291493be54.png)